docs(keycloak): Update documentation for Keycloak auth manager CLI usage and permission management (#54928)
mistercrunch/Airflow: providers/keycloak/docs/auth-manager/manage/permissions.rst

Manage user permissions with Keycloak auth manager
==================================================

Please setup your Keycloak instance and required parameters for the Keycloak auth manager.

> Setting up the permissions can easily done using CLI commands.

Reviewer: ```suggestion
Setting up the permissions can be done using CLI commands.
```

> Please setup your Keycloak instance and required parameters for the Keycloak auth manager.
> 
> Setting up the permissions can easily done using CLI commands.

Reviewer 2: ```suggestion
You can set-up and manage user permissions with Keycloak auth manager in different ways:

1. Using Airflow CLI
2. Using Keycloak console

With Airflow CLI
----------------
Setting up the permissions can easily done using CLI commands.
```

> It can create the permissions and needed resources easily.

Reviewer: ```suggestion
They can create the permissions and needed resources easily.
```

> There is two options to create the permissions:

Reviewer: ```suggestion
There are two options to create the permissions:
```

> * Create all permissions in one go using single CLI command

Reviewer: ```suggestion
* Create all permissions in one go using one single CLI command
```

> * Create step by step using the CLI commands

Reviewer: ```suggestion
* Create step-by-step using the CLI commands
```

What is this line supposed to mean? It creates a step-by-step (of what?) for the user??

Author: I forgot to add permissions there, and also included what is created underneath in parentheses.

> CLI commands taking following parameters:

Reviewer: ```suggestion
CLI commands take the following parameters:
```

* ``--username``: Keycloak admin username
* ``--password``: Keycloak admin password
* ``--user-realm``: Keycloak user realm (default: master)
* ``--client-id``: Keycloak client id (default: admin-cli)

> Please check the documentation for more information about the parameters.
> 
> `Keycloak auth manager CLI </cli-refs.html>`_

Reviewer: ```suggestion
Please check the `Keycloak auth manager CLI </cli-refs.html>`_ documentation for more information about accepted parameters.
```

> One go creation of permissions

Reviewer: ```suggestion
One-go creation of permissions
```

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

There is a single command do all the magic for you.

.. code-block:: bash

  airflow keycloak create-all

This will create scopes, resources and permissions in one go.

> Step by step creation of permissions

Reviewer: ```suggestion
Step-by-step creation of permissions
```

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> First we need to create scopes for the permissions. These scopes later will be used in Keycloak authorization [1].

Reviewer: This should use RestructuredText’s footnote syntax. https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#footnotes

Author: Updated with the RestructuredText’s footnote syntax.

.. code-block:: bash

  airflow keycloak create-scopes

> This will create scopes for certain types of permissions.

Reviewer: It would be more idiomatic to put the description _before_ the command instead so the user knows what’s going to happen before typing in the command. (Especially since the descriptions use future tense.)

Author: Updated and moved them before the command. Agree, nice one

.. code-block:: bash

  airflow keycloak create-resources

This will create resources for certain types of permissions.

.. code-block:: bash

  airflow keycloak create-permissions

Finally we create the permissions using the previously created scopes and resources.

This will create

* read-only permissions
* admin permissions
* user permissions
* operations permissions

More resources about permissions can be found in the official documentation of Keycloak:

1- `Keylcloak Authorization Process <https://www.keycloak.org/docs/latest/authorization_services/index.html#the-authorization-process>`_

2- `Keycloak Permission Overview <https://www.keycloak.org/docs/latest/authorization_services/index.html#_permission_overview>`_

> 3- `Keycloak Creating scope-based Permissions <https://www.keycloak.org/docs/latest/authorization_services/index.html#_policy_overview>`_

Reviewer: Would these handle correctly in rst?

Author: Yes, I found them from the code base :) https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#hyperlinks

<img width="1918" height="943" alt="Screenshot from 2025-08-26 20-00-41" src="https://github.com/user-attachments/assets/0686bdef-4514-40cf-bb0f-69c79fe844cb" />
